ДЗ13 готово

EvgeniiKovalev opened the pull request:
Прошу ответов на вопросы в коде

EvgeniiKovalev committed:
ДЗ13 готово

sproshchaev commented on src/main/java/ru/otus/java/basic/homeworks/homework13/Human.java:
```diff
@@ -0,0 +1,106 @@
+                power -= (float) distance * powerConsumption / 100;
+                return true;
+            }
+            return false;
```
Здесь можно использовать более компактную проверку, чтобы уменьшить вложенность условий
```
if (currentTransport instanceof Bicycle && power < (float) distance * powerConsumption / 100) {
    System.out.printf("У человека не хватает сил чтобы проехать на велосипеде расстояние %d\n", distance);
    return false;
}
if (currentTransport != null && currentTransport.move(area, distance)) {
    if (currentTransport instanceof Bicycle) {
        power -= (float) distance * powerConsumption / 100;
    }
    return true;
}

```

EvgeniiKovalev replied:
Проверил, тесты показали что логика поломалась. А вообще получается что цикломатическую сложность вы звменяете на логическую. Ваш вариант мне не понятен с логической точки зрения

sproshchaev commented on src/main/java/ru/otus/java/basic/homeworks/homework13/TransportFactory.java:
```diff
@@ -0,0 +1,23 @@
+                transport = new Offroader();
+                break;
+        }
+        return transport;
```
еще есть компактная форма switch с ->
```
return switch (type) {
    case CAR -> new Car();
    case HORSE -> new Horse();
    case BICYCLE -> new Bicycle();
    case OFFROADER -> new Offroader();
};

```

EvgeniiKovalev replied:
Спасибо, в java версии >= 12 такой формат switch case возможен, но текущий проект на версии java = 11

EvgeniiKovalev committed:
ДЗ13 пофиксил switch case

FlameXander commented on src/main/java/ru/otus/java/basic/homeworks/homework13/Human.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    public boolean sitDown(TransportType transportType) {
+        if (this.currentTransport == null) {
```
```
if (this.currentTransport != null) {
  return false;
}
currentTransport = transportFactory.createTransport(transportType);
return true;
```

EvgeniiKovalev replied:
переписал решение, применил принцип, сначала проверяю негативное условие и  выход, потом код для позитивного условия

EvgeniiKovalev committed:
ТЗ17 добавлена ссылка на водителя в транспорте.
Удалено использование интерфейса как ненужное.
Для проверки на перемещение по разрешенной территории в класс транспорта добавлен список запрещенных территорий.
Логика уменьшения энергии ездока на велосипеде перенесена из человека в велосипед

EvgeniiKovalev committed:
ДЗ17 применил автом. форматирование и в Human.getUp сначала проверяется негативный вариант
